src/DotNetty.Transport/Channels/IChannelHandlerContext.cs

```diff
@@ -67,11 +67,13 @@ public interface IChannelHandlerContext : IAttributeMap
 
         IChannelHandlerContext Read();
 
-        Task WriteAsync(object message); // todo: optimize: add flag saying if handler is interested in task, do not produce task if it isn't needed
+        ValueTask WriteAsync(object message); // todo: optimize: add flag saying if handler is interested in task, do not produce task if it isn't needed
 
         IChannelHandlerContext Flush();
 
         Task WriteAndFlushAsync(object message);
```

Comment:
Since WriteAndFlushAsync is consequent WriteAsync and Flush, and later can cause former to complete synchronously, ValueTask from WriteAsync backend by recyclable future is always materialized in full blown Task. So I kept Task as return type. Assumption is that those who want to do write+flush in non-allocating fire-and-forget mode  call overload and pass _false_ in notifyComplete. In that case overload always returns completed ValueTask.